packages/metalk8s-sosreport/src/metalk8s.py

```diff
@@ -0,0 +1,119 @@
+#! /bin/env python
+
+from sos.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+from os import path
+
+
+class metalk8s(Plugin, RedHatPlugin, UbuntuPlugin):
+
+    """Metalk8s plugin"""
+
+    plugin_name = 'metalk8s'
+    packages = ('kubernetes', 'kubernetes-master')
+    profiles = ('container',)
+    files = ('/etc/kubernetes/admin.conf',)
+
+    option_list = [
+        ('all', 'also collect all namespaces output separately',
+            'slow', False),
+        ('describe', 'capture descriptions of all kube resources',
+            'fast', False),
+        ('podlogs', 'capture logs for pods', 'slow', False),
+    ]
+
+    def check_is_master(self):
+        return any([path.exists("/etc/kubernetes/admin.conf")])
+
+    def setup(self):
+        self.add_copy_spec('/etc/kubernetes')
+        self.add_copy_spec('/var/log/pods')
+        self.add_copy_spec('/var/log/metalk8s-bootstrap.log')
+
+        services = [
+            'kubelet',
+        ]
+
+        for service in services:
+            self.add_journal(units=service)
+
+        # We can only grab kubectl output from the master
+        if self.check_is_master():
+            kube_cmd = 'kubectl '
+            if path.exists('/etc/kubernetes/admin.conf'):
+                kube_cmd += '--kubeconfig=/etc/kubernetes/admin.conf'
+
+            kube_get_cmd = 'get -o json '
+            for subcmd in ['version', 'config view']:
+                self.add_cmd_output('{0} {1}'.format(kube_cmd, subcmd))
+
+            # get all namespaces in use
+            namespaces_result = self.get_command_output('{0} get namespaces'.format(kube_cmd))
+            kube_namespaces = [n.split()[0] for n in namespaces_result['output'].splitlines()[1:] if n]
+
+            resources = [
+                'pods',
+                'deploy',
+                'rc',
+                'services',
+                'ds',
+                'cm',
+            ]
+
+            # nodes and pvs are not namespaced, must pull separately.
+            # Also collect master metrics
+            self.add_cmd_output([
+                '{} get -o json nodes'.format(kube_cmd),
+                '{} get -o json pv'.format(kube_cmd),
+                '{} get --raw /metrics'.format(kube_cmd)
+            ])
+
+            for n in kube_namespaces:
+                kube_namespace = '--namespace={}'.format(n)
+                if self.get_option('all'):
+                    kube_namedspaced_cmd = '{0} {1} {2}'.format(kube_cmd, kube_get_cmd, kube_namespace)
```

Comment:
minor (also on other lines):

```suggestion
                    kube_namespaced_cmd = '{0} {1} {2}'.format(kube_cmd, kube_get_cmd, kube_namespace)
```